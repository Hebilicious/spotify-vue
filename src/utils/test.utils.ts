import { VueWrapper } from '@vue/test-utils';
import { expect, it } from 'vitest';

export const testSnapShot = (wrapperGetter: () => VueWrapper) => {
  it('Matches the snapshot', () => {
    expect(wrapperGetter().html()).toMatchSnapshot();
  });
};
