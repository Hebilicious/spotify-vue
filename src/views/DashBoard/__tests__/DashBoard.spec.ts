import { testSnapShot } from "@/utils/test.utils";
import { createTestingPinia } from "@pinia/testing";
import { shallowMount } from "@vue/test-utils";
import { describe, fn } from "vitest";
import DashBoardVue from "../DashBoard.vue";

describe('DashBoard', () => {
    testSnapShot(() => shallowMount(DashBoardVue, {
        global: {
            plugins: [createTestingPinia({ createSpy: () => fn() })]
        }
    }))
})