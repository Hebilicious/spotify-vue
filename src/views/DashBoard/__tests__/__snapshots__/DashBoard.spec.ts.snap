// Vitest Snapshot v1

exports[`DashBoard > Matches the snapshot 1`] = `
"<div class="home-view">
  <loader-stub></loader-stub>
</div>"
`;
